### ISSUE TITLE
[v1.7.0-rc.0] Typing queries causes type issues with selectors

### ISSUE THREAD

**Issue by drewgaze** [reduxjs/redux-toolkit#1817]
I upgraded my app to v1.7.0-rc.0 and had selectors which were previously working gain type issues. It looks like selectors which are created based on the endpoints' selectors type state as never, which causes selectors composed from them to error in typescript due to lack of overlapping types for state.

Here is a sandbox which reproduces the issue: https://codesandbox.io/s/rtk-query-selector-typing-issue-lzs3r?file=/src/api.ts

**Comment by markerikson** [comment]
Yep, we had a couple other reports as well. This appears to be related to the TS typing work we've been doing in Reselect 4.1.x.

I was doing some investigation last night and was some some very odd behavior from TS itself that is confusing me.  I'll keep digging further.  Thanks for the additional repro, that should be helpful!

**Comment by markerikson** [comment]
So from what I can see this appears to be A) sort of coming from Reselect, but B) actually a TS issue somehow.

I've narrowed it down to the behavior in https://codesandbox.io/s/typescript-playground-export-forked-ipvpi?file=/index.ts when using RTK 1.7.0-rc.0 and Reselect 4.1.5.  I'll paste the source here:

```ts
import { createSelector } from "reselect";
import { createSlice, configureStore } from "@reduxjs/toolkit";
import { createApi, fetchBaseQuery, retry } from "@reduxjs/toolkit/query/react";

const baseQuery = retry(
  fetchBaseQuery({ baseUrl: "https://jsonplaceholder.typicode.com" }),
  { maxRetries: 5 }
);

interface Todo {
  userId: number;
  id: number;
  title: string;
  completed: boolean;
}

type Todos = Array<Todo>;

const exampleApi = createApi({
  reducerPath: "api",
  baseQuery,
  endpoints: (build) => ({
    getTodos: build.query<Todos, string>({
      query: () => "/todos"
    })
  })
});

// Create a Reselect-based selector for reading the query state

const exampleQuerySelector = exampleApi.endpoints.getTodos.select("/");

// Now we'll try to introspect the selector's arguments
type eqs1 = typeof exampleQuerySelector;
/*
This is fine:

type p1 = [state: RootState<{
    getTodos: QueryDefinition<string, BaseQueryFn<string | FetchArgs, unknown, FetchBaseQueryError, StaggerOptions, {}>, never, Todos, string>;
}, never, Todos>]
*/
type p1 = Parameters<eqs1>;
// But this results in `type p10 = {}` ??!?!?!!?
type p10 = p1[0];

// Actual user repro code below, which also fails in the Reselect type tests:

const todosSelector = createSelector([exampleQuerySelector], (queryState) => {
  return queryState?.data?.[0] ?? ({} as Todo);
});

// This errors because the `state` arg is `never`
const firstTodoTitleSelector = createSelector(
  [todosSelector],
  (todo) => todo?.title
);

```

**Comment by markerikson** [comment]
Good news - this is actually a real RTKQ types bug, and we've got it fixed!  PR momentarily...

**Comment by markerikson** [comment]
Merged and will be out in whatever the next 1.7 release is (`rc.1` or final)
